### ISSUE TITLE
Enable taking node temporarily offline due to specific machine issue in Adoptium

### ISSUE THREAD

**Issue by sophia-guo** [adoptium/aqa-tests#5730]
Adding the parameter SLACK_CHANNEL to the configuration of https://ci.adoptium.net/view/Test_grinder/job/Test_Job_Auto_Gen/ can take node offline due to specfiic machine issues. 

This issue opened to monitor any issues with this enabled.

**Comment by sophia-guo** [comment]
[test-azure-ubuntu2404-x64-1](https://ci.adoptium.net/computer/test-azure-ubuntu2404-x64-1) was hit twice due to the `No space left on device`. It was not marked as offline as `No space left on device` was on the error lists  https://github.com/adoptium/aqa-tests/pull/5731. 

https://ci.adoptium.net/job/Test_openjdk21_hs_sanity.openjdk_x86-64_linux_testList_1/19/console

```
15:44:39  Exception: hudson.AbortException: Failed to run ssh-agent: mkdtemp: private socket dir: No space left on device
15:44:39  
[Pipeline] timeout
```

https://ci.adoptium.net/job/Test_openjdk21_hs_special.system_x86-64_linux/28/console

```
[Pipeline] echo
15:37:04  Exception: hudson.AbortException: Failed to run ssh-agent: mkdtemp: private socket dir: No space left on device
15:37:04  
```

Currently [test-azure-ubuntu2404-x64-1](https://ci.adoptium.net/computer/test-azure-ubuntu2404-x64-1)  is marked offline. I believe it's marked offline by jenkins auto-offline machines [that are low on space?@sxa is it marked offline by infra's scheduled task?](https://adoptium.slack.com/archives/C53GHCXL4/p1730289619020629?thread_ts=1730227347.646299&cid=C53GHCXL4). How would infra process this case?

**Comment by adamfarley** [comment]
Heya @sophia-guo 

FYI: It looks like the auto-offline logic isn't working at the moment. I'll raise an issue to cover this.

In short, jobs like [this one](https://ci.adoptium.net/job/Test_openjdk17_hs_sanity.openjdk_x86-64_linux_testList_1/24/console) fail due to lack of space, and our attempt to take the machine offline fails with this error:

```
Scripts not permitted to use staticMethod hudson.model.User current. Administrators can decide whether to approve or reject this signature.
```

Which I presume is being caused by [this code](https://github.com/adoptium/aqa-tests/blob/4ee3f9d5362811825a767c80046d1b9b0ecbf7dd/buildenv/jenkins/openjdk_tests#L573).

**Comment by llxia** [comment]
re https://github.com/adoptium/aqa-tests/issues/5730#issuecomment-2532012520, this is not the code issue. As the error stated, Adoptium Jenkins Admin needs to permit using staticMethod hudson.model.User current at Adoptium Jenkins.

**Comment by adamfarley** [comment]
The problem is that the code is attempting to use a method it is not authorized to do so.

You are correct in that one solution is to get Jenkins admins to authorise use of that static method.

Your solution also looks like the best one when I compared it to alternatives (such as using SimpleOfflineCause instead of UserCause, however I'm not seeing a trivial way to create instances of the Localisable class.

P.S. I also discovered that the setTemporarilyOffline method is deprecated in favour of setTemporaryOfflineCause. I'm noting that here in case this setTemporarilyOffline is removed in a future update.

**Comment by smlambert** [comment]
I have permitted it, but have also done so for a different method.  Not sure what other ones will pop up.  May be worth bringing that machine back online and sending a job to it to see if we get past any other approvals needed.

**Comment by adamfarley** [comment]
Will keep an eye open for automatic machine disabling in future triage (whether it works or not).

**Comment by sophia-guo** [comment]
https://ci.adoptium.net/view/Test_grinder/job/Grinder/12049/  same agent no space left again. [test-azure-ubuntu2404-x64-1](https://ci.adoptium.net/computer/test-azure-ubuntu2404-x64-1). Just added the SLACK_CHANNEL parameter, so the permission issue happens again.  https://ci.adoptium.net/view/Test_grinder/job/Grinder/12049/ 

```
Also:   org.jenkinsci.plugins.workflow.actions.ErrorAction$ErrorId: 1e6db258-b702-4687-9fc1-2b21e188b8f8
org.jenkinsci.plugins.scriptsecurity.sandbox.RejectedAccessException: Scripts not permitted to use staticMethod hudson.model.User current
	at PluginClassLoader for script-security//org.jenkinsci.plugins.scriptsecurity.sandbox.whitelists.StaticWhitelist.rejectStaticMethod(StaticWhitelist.java:258)
```

@smlambert maybe you can check if permission message pops up? And after this fix if we don't like this feature available for grinder I can remove the parameter. 

@adamfarley if the machine is back with fix it's weird that it runs out of space in such short time.

**Comment by adamfarley** [comment]
> @adamfarley if the machine is back with fix it's weird that it runs out of space in such short time.

Agreed. Here's the issue: https://github.com/adoptium/infrastructure/issues/3843

Note that this was the second time in a month that this machine has run out of space and been resurrected as "fixed", so perhaps a lack of overall storage space isn't the problem.

**Comment by adamfarley** [comment]
Useful links:

API for "setTemporarilyOffline": https://javadoc.jenkins.io/hudson/model/Computer.html#setTemporarilyOffline(boolean,hudson.slaves.OfflineCause)

If we do go for a code fix instead of working around the UserLooks like this would be the easiest code fix:

```
currentNode.setTemporaryOfflineCause(true, new hudson.slaves.OfflineCause.ByCLI("${message}"))
```

ByCLI can also be replaced by "LaunchFailed" if we want to, with the rest of the code staying the same.
